db/db_impl/db_impl_write.cc

```diff
@@ -189,23 +238,18 @@ Status DBImpl::PebbleWriteImpl(const WriteOptions& write_options,
     stats->AddDBStats(InternalStats::kIntStatsNumKeysWritten, total_count);
     RecordTick(stats_, NUMBER_KEYS_WRITTEN, total_count);
 
-    ColumnFamilyMemTablesImpl column_family_memtables(
-        versions_->GetColumnFamilySet());
-    writer.status = WriteBatchInternal::InsertInto(
-        &writer, writer.sequence, &column_family_memtables, &flush_scheduler_,
-        write_options.ignore_missing_column_families, 0 /*log_number*/, this,
-        true /*concurrent_memtable_writes*/);
+    while (writer.ConsumeOne());
+    MultiBatchWriteCommit(writer.request);
 
     WriteStatusCheck(writer.status);
-
     if (!writer.FinalStatus().ok()) {
       writer.status = writer.FinalStatus();
     }
-    PebbleWriteCommit(writer.request);
   } else if (writer.request->commit_lsn != 0) {
-    PebbleWriteCommit(writer.request);
+    writer.multi_batch_writer.pending_wb_cnt.store(0);
```

Comment:
It's possible that a write with !status.ok() can still be written to memtable by another writer.

Reply:
Yes It's possible.

Reply:
I think this is dangerous. How about let leader reset the failed writer's `pending_wb_cnt`?

Reply:
Sure